### PR TITLE
SOL-112355: Integration tests to cover the processing of requests and responses

### PR DESCRIPTION
**Pull request by oodigie**
**Changes:**
- SOL-112354: Start and termination of a RequestReplyMessageReceiver integration tests
- feat: updated the tests to fix potential race conditions with the use of time.Millisecond
- feat: added an assertion to check the error returned from replier.Reply in one of the tests
- feat: fixed the failing tests to check that error returns from replier.Reply after service termination
- SOL-112355: Integration tests to cover the requests and responses processing using the RequestReplyMessagePublisher and RequestReplyMessageReceiver
- SOL-112355: updated the integration tests to provide a more consistent test outcome

### CODE REVIEW

File: test/request_reply_message_receiver_test.go [hunk 1]
```diff
@@ -34,81 +40,992 @@ const subscriptionString = "some-subscription"
...
+					client := helpers.GetClient(messagingService)
+					Expect(client.DataTxMsgCount).To(Equal(int64(3)))                                                // 1 direct, 1 request and 1 reply message                                            // requests & replies
+					Expect(messagingService.Metrics().GetValue(metrics.DirectMessagesReceived)).To(Equal(uint64(3))) // 3 messages
+				})
+
+				It("should properly handle direct massages published to request-reply topic with the ReceiveAsync() function", func() {
+					rRMessagesCount := uint64(0)
+					directMessagesCount := uint64(0)
+
+					// apparently the test degrades as the number of published messages increases
```
Comment (cjwmorgan-sol) [inline test/request_reply_message_receiver_test.go:977]: This is due to the receiver discarding received messages when the number of buffered messages exceeds 50 messages. The feature Branch SOL-62455 has added receiver back pressure configuration method on builder to provide easy access to configure option for the receiver buffering and drop behaviour. Note this was configurable with the receiver sting properties before as well.

This test publishes over 50 messages so its possible to have discard issues here. Either change the number of published messages to 50 the default receiver buffer size or configure the receiver with a buffer size of 100 or greater or have your test account for the discarded messages via discard indication on the received messages.

**Review by cjwmorgan-sol — CHANGES_REQUESTED** [review]
There is a failing test due to a receiver configuration issue.
@oodigie please address it.

**Comment by cjwmorgan-sol** [comment]
With PR #43 merged this PR might need to sync with the base feature branch SOL-62455 now.

File: test/request_reply_message_receiver_test.go (hunk 1 above)
Comment (oodigie) [reply test/request_reply_message_receiver_test.go:977]: Okay, thanks for the explanation. I will update the test case and confirm.

**Comment by oodigie** [comment]
@cjwmorgan-sol,
I am currently publishing 1000 messages for these tests and I also configured the receiver back pressure to have a buffer of 1000 messages (to accommodate these messages). The two test failures are due to timeout waiting for the reply for the published messages - **"Timed out waiting to receive reply message"**. Apparently I do not see this failures on my local machine so it may be due to stress of the test nodes and/or available compute resource.

Seeing as the average time to typically receive a reply per message is about < 1 second, probably < 100 milliseconds (changes per version of the broker), I will either have to increase the timeout to allow receiving all the replies or reduce the amount of messages being published. I will reduce the total published message to 500.

**Review by cjwmorgan-sol — APPROVED** [review]
All check passing in git actions, and the only jenkins failure were independent of these changes (all known intermittent).
This looks good to me.
And is ready for merge.

**Pull request merged**